forc/src/utils/helpers.rs

```diff
@@ -52,10 +53,73 @@ pub fn read_manifest(manifest_dir: &Path) -> Result<Manifest, String> {
             ))
         }
     };
-    match toml::from_str(&manifest) {
+
+    let manifest = match toml::from_str(&manifest) {
         Ok(o) => Ok(o),
         Err(e) => Err(format!("Error parsing manifest: {}.", e)),
+    }?;
+
+    validate_manifest(manifest)
+}
+
+// Using (https://github.com/rust-lang/cargo/blob/master/src/cargo/util/toml/mod.rs#L570 +
+// https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_new.rs#L166) for reference
```

Comment:
These should be permalinks. You can press "y" on your keyboard when you have the code page open and it'll convert it into a permalink in the address bar.

Reply:
> Looks fine to me, will defer to @mitchmindtree and @JoshuaBatty

woops didn't see this, sorry to request you again